main/forum/editthread.php

```diff
@@ -0,0 +1,220 @@
+$cId = isset($courseInfo['real_id']) ? $courseInfo['real_id'] : false; 
+
+/* Is the user allowed here? */
+
+// The user is not allowed here if:
+
+// 1. the forumcategory or forum is invisible (visibility==0) and the user is not a course manager
+if (!api_is_allowed_to_edit(false, true) &&
+    (($current_forum_category['visibility'] && $current_forum_category['visibility'] == 0) || $current_forum['visibility'] == 0)
+) {
+    api_not_allowed();
+}
+
+// 2. the forumcategory or forum is locked (locked <>0) and the user is not a course manager
+if (!api_is_allowed_to_edit(false, true) &&
+    (($current_forum_category['visibility'] && $current_forum_category['locked'] <> 0) OR $current_forum['locked'] <> 0)
+) {
+    api_not_allowed();
+}
+
+// 3. new threads are not allowed and the user is not a course manager
+if (!api_is_allowed_to_edit(false, true) &&
+    $current_forum['allow_new_threads'] <> 1
+) {
+    api_not_allowed();
+}
+// 4. anonymous posts are not allowed and the user is not logged in
+if (!$_user['user_id'] AND $current_forum['allow_anonymous'] <> 1) {
+    api_not_allowed();
+}
+
+// 5. Check user access
+if ($current_forum['forum_of_group'] != 0) {
+    $show_forum = GroupManager::user_has_access(
+        api_get_user_id(),
+        $current_forum['forum_of_group'],
+        GroupManager::GROUP_TOOL_FORUM
+    );
+    if (!$show_forum) {
+        api_not_allowed();
+    }
+}
+
+// 6. Invited users can't create new threads
+if (api_is_invitee()) {
+    api_not_allowed(true);
+}
+
+$groupId = api_get_group_id();
+if (!empty($groupId)) {
+    $group_properties = GroupManager :: get_group_properties($groupId);
+    $interbreadcrumb[] = array('url' => '../group/group.php?'.api_get_cidreq(), 'name' => get_lang('Groups'));
+    $interbreadcrumb[] = array('url' => '../group/group_space.php?'.api_get_cidreq(), 'name' => get_lang('GroupSpace').' '.$group_properties['name']);
+    $interbreadcrumb[] = array('url' => 'viewforum.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']), 'name' => $current_forum['forum_title']);
+    $interbreadcrumb[] = array('url' => 'newthread.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']),'name' => get_lang('EditThread'));
+} else {
+    $interbreadcrumb[] = array('url' => 'index.php?'.api_get_cidreq(), 'name' => $nameTools);
+    $interbreadcrumb[] = array('url' => 'viewforumcategory.php?'.api_get_cidreq().'&forumcategory='.$current_forum_category['cat_id'], 'name' => $current_forum_category['cat_title']);
+    $interbreadcrumb[] = array('url' => 'viewforum.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']), 'name' => $current_forum['forum_title']);
+    $interbreadcrumb[] = array('url' => '#', 'name' => get_lang('EditThread'));
+}
+
+/* Resource Linker */
+
+if (isset($_POST['add_resources']) AND $_POST['add_resources'] == get_lang('Resources')) {
+    $_SESSION['formelements']	= $_POST;
+    $_SESSION['origin']			= $_SERVER['REQUEST_URI'];
```

Comment:
Very dangerous. Don't put $_SERVER variables into $_SESSION unless you're sure that it's filtered. In this case, add a comment saying "// Filtered below" or something like that.